python/triton/experimental/gluon/language/_semantic.py

```diff
@@ -14,6 +17,74 @@ class GluonSemantic(TritonSemantic[ttgl.tensor]):
     def __init__(self, builder: GluonOpBuilder):
         self.builder = builder
 
+    def _broadcast_shapes(self, lhs_shape: List[int], rhs_shape: List[int]):
+        if len(lhs_shape) != len(rhs_shape):
+            raise ValueError(f"Cannot broadcast, rank mismatch: {lhs_shape}, {rhs_shape}")
+
+        ret_shape = []
+        for i, left in enumerate(lhs_shape):
+            right = rhs_shape[i]
+            if left == 1:
+                ret_shape.append(right)
+            elif (right == 1) or (right == left):
+                ret_shape.append(left)
+            else:
+                raise ValueError("Cannot make_shape_compatible: incompatible dimensions "
+                                 "at index " + str(i) + ": " + str(left) + " and " + str(right))
+        return ret_shape
+
+    def expand_dims(self, input: TensorTy, axis: int) -> TensorTy:
+        dst_shape = [ttgl._unwrap_if_constexpr(x) for x in input.shape]
+        dst_shape.insert(axis, 1)
+
+        if axis < 0:
+            axis += len(input.shape)
+
+        assert isinstance(input.type, ttgl.distributed_type)
+        layout = input.type.layout
+        assert isinstance(layout, SliceLayout)
+        assert layout.dim == axis
```

Comment:
Do you mind adding some error messages here? In particular, the error message printing the invalid value I find is super helpful because often it's hard to tell by looking at the kernel code (it's like writing C++ but everything is `auto`)